src/Workspaces/Core/Portable/Shared/Extensions/ContextQuery/AbstractSyntaxContext.cs

```diff
@@ -80,21 +82,27 @@ private ISet<INamedTypeSymbol> ComputeOuterTypes(CancellationToken cancellationT
                 var containingType = enclosingSymbol.GetContainingTypeOrThis();
                 if (containingType != null)
                 {
-                    return containingType.GetContainingTypes().ToSet();
+                    return new Tuple<ISet<ITypeSymbol>, ITypeSymbol>(containingType.GetContainingTypesAndThis().ToSet(), containingType);
                 }
             }
 
-            return SpecializedCollections.EmptySet<INamedTypeSymbol>();
+            return new Tuple<ISet<ITypeSymbol>, ITypeSymbol>(SpecializedCollections.EmptySet<ITypeSymbol>(), null);
         }
 
-        public ISet<INamedTypeSymbol> GetOuterTypes(CancellationToken cancellationToken)
+        public IEnumerable<ITypeSymbol> GetOuterTypes(bool includeContainingType, CancellationToken cancellationToken)
```

Comment:
This flag seems a bit weird. What's wrong with just letting the caller call GetContainingTypeOrThis() and then GetOuterTypes()? GetOuterTypes now seems really convoluted.